src/lib/iproute2_cmdgen.c

```diff
@@ -10,12 +10,251 @@
  *
  * Copyright (C) 2024 Okda Networks, <aaqrbaw@okdanetworks.com>
  */
+#include <ctype.h>
 
 #include "iproute2_cmdgen.h"
 
+#define CMD_LINE_SIZE 1024
+
+typedef enum {
+    ADD,
+    DELETE,
+    UPDATE,
+    UNKNOWN,
+} oper_t;
+
+
+void copy_argv(char ***dest, char **src, int argc)
+{
+    *dest = (char **) malloc(argc * sizeof(char *));
+    if (*dest == NULL) {
+        fprintf(stderr, "Memory allocation failed\n");
+        exit(EXIT_FAILURE);
+    }
+    for (int i = 0; i < argc; i++) {
+        (*dest)[i] = strdup(src[i]);
+        if ((*dest)[i] == NULL) {
+            fprintf(stderr, "Memory allocation failed\n");
+            exit(EXIT_FAILURE);
+        }
+    }
+}
+
+
+void free_argv(char **argv, int argc)
+{
+    // Free memory for each string
+    for (int i = 0; i < argc; i++) {
+        free(argv[i]);
+    }
+    // Free memory for the array of pointers
+    free(argv);
+}
+
+/*
+ * if input is "iproute2-ip-link:dummy" it return "dummy"
+ */
+char *strip_yang_iden_prefix(const char *input)
+{
+    const char *colon_pos = strchr(input, ':');
+    if (colon_pos != NULL) {
+        // Allocate memory for output string
+        char *output = malloc(strlen(colon_pos + 1) + 1);
+        if (output == NULL) {
+            fprintf(stderr, "Memory allocation failed\n");
+            exit(EXIT_FAILURE);
+        }
+        strcpy(output, colon_pos + 1);
```

Comment:
could strlcpy() be used ?

Reply:
won't that add a dependency ? as it seems that this is not added by default in libc, on my machine it's not available.